src/client/Microsoft.Identity.Client/Http/NoRetryPolicy.cs

```diff
@@ -2,20 +2,16 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
 
 namespace Microsoft.Identity.Client.Http
 {
     internal class NoRetryPolicy : IRetryPolicy
     {
-        public int DelayInMilliseconds { get => throw new NotImplementedException(); }
-
-        public bool pauseForRetry(HttpResponse response, Exception exception, int retryCount)
+        public Task<bool> pauseForRetryAsync(HttpResponse response, Exception exception, int retryCount, ILoggerAdapter logger)
         {
-            return false;
+            throw new NotImplementedException();
```

Comment:
Revert this bacl to return false. This will be used in case the user is passing own http factory and want to have their own retry logic

Reply:
return `Task.FromResult(false)` because you are returning a task (a promise in JS terms)

Reply:
done